src/fu-util.c

```diff
@@ -2973,6 +2973,17 @@ main (int argc, char *argv[])
 		return EXIT_FAILURE;
 	}
 
+	/* send our implemented feature set */
+	if (!fwupd_client_set_feature_flags (priv->client,
+					     FWUPD_FEATURE_FLAG_CAN_REPORT |
+					     FWUPD_FEATURE_FLAG_UPDATE_ACTION |
+					     FWUPD_FEATURE_FLAG_DETACH_ACTION,
+					     priv->cancellable, &error)) {
+		g_printerr ("Failed to set front-end features: %s\n",
+			    error->message);
+		return EXIT_FAILURE;
+	}
```

Comment:
To add a few other dimensions of of complexity to your thought process that found motd:
* the other non-obvious implication is how the command line client behaves if it has a tty or not.  In non-interactive mode you wouldn't be able to see those messages for detach or unplug anymore.

* The message to unplug, it might not actually be _needed_.  IE it's just to start the boot into the new firmware, or to finalize a firmware upgrade.  If you want to be able to support fully automated updates for stuff that has the flag `usable-during-update` you would probably put in the XML `LVFS::UpdateMessage`, and `LVFS::UpdateImage` but not put `update-action` in client requirements.

Reply:
>  In non-interactive mode you wouldn't be able to see those messages for detach or unplug

I didn't think about that. I guess we can just set the flags correctly if `isatty==0`

> you would probably put in the XML LVFS::UpdateMessage, and LVFS::UpdateImage but not put update-action in client requirements

Right, this is why I didn't "auto require" the features when `LVFS::UpdateMessage` was set.

Reply:
> I didn't think about that. I guess we can just set the flags correctly if isatty==0

That's what I did. I think it's a lot nicer with FuEngineRequest. Comments welcome!

Reply:
I think you only got it done for `fu-tool.c`, forgot for `fu-util.c`